Cargo.toml

```diff
@@ -11,7 +11,38 @@ edition = "2018"
 util = [ "structopt", "driver-pal/hal", "simplelog", "humantime" ] 
 default = [ "util", "serde" ]
 
+# features for examples
+stm32f0xx = []
```

Comment:
is there value in having -all- of these example targets? i am reasonably confident that building for an m0/m3/m4 would cover most of our bases / simplify a few things?

Reply:
This depends on what you mean by "cover most of our bases".  If you mean "cover most potential hobby users" then I think you will get over half with bluepill, and well over 90% with bluepill, blackpill-stm32f411, and discovery-stm32f303.

Trying to learn Rust I've spent a lot of time  looking at examples that appeared to be current but then found they only worked on some old version of libraries and with only one specific MCU. So what I have being trying to do is verify/illustrate the "hardware abstraction layer" aspect of crates. That way a user like me can see how to develop application code without a total commitment to what hardware it will run on. For that, one would like more rather than fewer setups. For me, with a general coding background but no rust or embedded experience, the application code has been fairly straight forward. It has been the setup code and the `Cargo.toml` that are hard. (It is a lot better with impl traits working.)

Of course, ideally one would like `setup()` to be as small as possible, with as much as possible being generic code..

Then if by "cover most of our bases" you mean coverage as in integration testing, more is always better. And it seems to be not only the target triple but also the MCU that needs to be tested. However, it is the run testing rather than build testing that is a lot of work. Part of my hope is that if other users can see that the code actually builds for their platform then they will do run testing.

I am intending to also build these at https://pdgilbert.github.io/eg_stm_hal/#table-of-additional-examples-status. There the idea is to use release crates but update with git versions of hals as they evolve. Here, release versions of hals are used, but of course examples need to be updated if the crate's code changes. So examples stay current with git versions of the crate. But in the next period of change toward `e-h 1.0.0` I think both sides will be changing so quickly that the distinction doesn't matter much.

So all that is to explain my rationale, but let me know what you prefer. My own preference order would be
 1/ leave all the setup code in but try to separate into a sourced file and run CI.
 2/  leave all the setup code in but try to separate into a sourced file and trim CI to a few MCUs.
 3/ trim out some of the setups but add a pointer to my test site.

In the end though I think this really is your choice, so let me know.

Reply:
I've removed all the example devices other than bluepill and blackpill stm32f411. (Also added `compat` feature to include/omit  embedded-hal-compat.)